src/proxy/topic/ReliableTopicListenerRunner.ts

```diff
@@ -36,16 +36,18 @@ export class ReliableTopicListenerRunner<E> {
         }
 
         this.ringbuffer.readMany(this.sequenceNumber, 1, this.batchSize).then((result: Array<RawTopicMessage>) => {
-            result.forEach((raw: RawTopicMessage) => {
-                var msg = new TopicMessage<E>();
-                msg.messageObject = this.serializationService.toObject(raw.payload);
-                msg.publisher = raw.publisherAddress;
-                msg.publishingTime = raw.publishTime;
-                setImmediate(this.listener, msg);
-                this.sequenceNumber++;
-            });
+            if (!this.cancelled) {
```

Comment:
This cancelled check is in the for loop on java implementation, can you check it?

Reply:
For Node.js it does not matter. If `this.cancelled` is false before going into `forEach` loop below, it will stay false until all items in `result` are processed because the loop will not be interrupted (single thread execution of node.js)